tests/commands/package/test_call.py

```diff
@@ -758,3 +762,174 @@ def test_already_packaged(package_command, first_app, tmp_path):
+        ("verify-app-template", "first"),
+        # App tools are verified for first app
+        ("verify-app-tools", "first"),
+        # Package the first app
+        (
+            "package",
+            "first",
+            {
+                "adhoc_sign": False,
+                "identity": None,
+            },
+        ),
+        # App template is verified for second app
+        ("verify-app-template", "second"),
+        # App tools are verified for second app
+        ("verify-app-tools", "second"),
+        # Package the second app
+        (
+            "package",
+            "second",
+            {
+                "adhoc_sign": False,
+                "identity": None,
+                "package_state": "first",
+            },
+        ),
+    ]
+
+
+def test_package_app_invalid(package_command, first_app, second_app):
+    """If an invalid app name is passed to --app, raise an error."""
+    # Add two apps
+    package_command.apps = {
+        "first": first_app,
+        "second": second_app,
+    }
+
+    # Configure the --app option with an invalid app
+    options, _ = package_command.parse_options(["--app", "invalid"])
+
+    # Running the command should raise an error
+    with pytest.raises(
+        BriefcaseCommandError,
+        match=r"App 'invalid' does not exist in this project.",
+    ):
+        package_command(**options)
+
+
+def test_package_app_duplicate(package_command, first_app):
+    """Passing --app multiple times with the same name should not double-package."""
+    # Add a single app
+    package_command.apps = {
+        "first": first_app,
+    }
+
+    # Configure --app with duplicates
+    options, _ = package_command.parse_options(["--app", "first", "--app", "first"])
+
+    # Run the package command
+    package_command(**options)
+
+    # The app should be packaged only once
+    assert (
+        package_command.actions.count(
+            ("package", "first", {"adhoc_sign": False, "identity": None})
+        )
+        == 1
+    )
+
+
+def test_package_app_none_defined(package_command):
+    """Packaging when no apps are defined should be a no-op."""
+    # No apps defined
+    package_command.apps = {}
+
+    # Configure no command line options
+    options, _ = package_command.parse_options([])
+
+    # Run the package command
+    result = package_command(**options)
+
+    # The right sequence of things will be done
+    assert result is None
+    assert package_command.actions == [
+        # Host OS is verified
+        ("verify-host",),
+        # Tools are verified
+        ("verify-tools",),
+    ]
+
+
+def test_package_with_update(package_command, first_app):
+    """If --update is passed, app is updated before packaging."""
+    package_command.apps = {"first": first_app}
+    options, _ = package_command.parse_options(["--app", "first", "--update"])
+    package_command(**options)
+
+    # Find the 'update' action for the app
+    update_action = next(
```

Comment:
It may be long-winded, but it's important to test the *entire* action sequence here, not just the one action that we think is important. The purpose of this test is to verify that Briefcase will perform the right sequence of actions - we can't leave anything up to assumed behavior here. We need to be certain that an updated package has everything updated (with the expected arguments), then built, then packaged.